src/components/user/loginModal.tsx

```diff
@@ -30,36 +31,52 @@ export const LoginModal: React.FC = () =>{
   }
 
   useEffect(()=>{
-    if(loggedIn){
-      window.location.href = "/";
-      setLoggedIn(false);
-    }
     if(alertDetail){
       window.setTimeout(()=>{setAlertDetail(false)},3000);
     }
+    if(loggedIn){
+      redirectToHome();
+    }
   }, [loggedIn, alertDetail])
 
-  function login(e){
+  const login = (e) => {
     e.preventDefault();
+    console.log(detail);
+    setLoginLoading(true);
     
     loginUserByEmail(detail)
     .then(res => {
-      if(res === "Unable to login"){
-        setAlertDetail(true);
-      }
-      else if(res === "error"){
-        setAlertDetail(true);
-      }
-      else{
+      if(res.status){
         setToken(res.token);
         localStorage.setItem('token', JSON.stringify(res.token));
+        setLoginLoading(false);
         setLoggedIn(true);
+      }else{
+        setAlertDetail(true);
+        setLoginLoading(false);
       }
     })
     
-    setDetail({email:'', password: ''});
+    setDetail(DefaultLogin);
   }
 
+  var string = I18n.t('loading');

```

Comment:
`string` is a keyword in Javascript/Typescript, should not be used as variable name